### PR TITLE
add proof loading to testvectors

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
testvectors now requires TESTING_PROOFS flag to run

Signed-off-by: Alexey-N-Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
Add proof loading before testvector.
Add TESTING_PROOFS flag to run testvectors.

### CODE REVIEW

File: test/core/test_vectors/test.cpp [hunk 1]
```diff
@@ -261,7 +262,20 @@ auto search() {
   return vectors;
 }
 
-struct TestVectors : testing::TestWithParam<MessageVector> {};
+struct TestVectors : testing::TestWithParam<MessageVector> {
+  static void SetUpTestCase() {
```
Comment (wer1st) [inline test/core/test_vectors/test.cpp:266]: In this case we always load proofs even if we already obtain them

File: test/core/CMakeLists.txt [hunk 2]
```diff
@@ -26,5 +26,9 @@ add_subdirectory(sector_storage)
 add_subdirectory(sectorblocks)
 add_subdirectory(serialization)
 add_subdirectory(storage)
-add_subdirectory(test_vectors)
+
+if (TESTING_PROOFS)
```
Comment (wer1st) [inline test/core/CMakeLists.txt:30]: Looks like TESTING_ACTORS is more suitable

File: test/core/test_vectors/test.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply test/core/test_vectors/test.cpp:266]: No, there is check if proofs has been already loaded in `ProofParamProvider::getParams`.

File: test/core/CMakeLists.txt (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply test/core/CMakeLists.txt]: Name has been changed.

**Review by turuslan — APPROVED** [review]

**Review by ortyomka — APPROVED** [review]

**Pull request merged**